<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An Interpretable Meta-clustering Framework for Single-cell 
    RNA-Seq Data Integration and Evaluation  • CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="An Interpretable Meta-clustering Framework for Single-cell 
    RNA-Seq Data Integration and Evaluation ">
<meta property="og:description" content="A workflow of (a) meta-clustering based on 
    inter-group similarity measures and (b) a ground-truth-free test metric to 
    assess the biological correctness of integration in real datasets. See Hu Z,
    Ahmed A, Yau C (2021) &lt;doi:10.1101/2021.03.29.437525&gt; for
    more details.">
<meta property="og:image" content="http://zhiyhu.github.io/CIDER/reference/figures/card.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CIDER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a>
    </li>
    <li>
      <a href="articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a>
    </li>
    <li>
      <a href="articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER)</a>
    </li>
    <li>
      <a href="articles/evaluation.html">Evaluating integration results by CIDER</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhiyhu/CIDER/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<<<<<<< Updated upstream
<div id="cider" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#cider" class="anchor"></a>CIDER</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>Clustering Single-cell RNA-Seq (scRNA-Seq) data from multiple samples or conditions are often challenged by confounding factors, such as batch effects and biologically relevant variability. Existing batch effect removal methods typically require strong assumptions on the composition of cell populations being near identical across samples. Here we present <strong>CIDER</strong>, a <strong>meta-clustering workflow</strong> based on inter-group similarity measures. The prototype of this method is firstly applied in <a href="https://www.sciencedirect.com/science/article/pii/S1535610820300428">Hu et al., Cancer Cell 2020</a>.</p>
=======
<div class="section level1">
<div class="page-header"><h1 id="cider-meta-clustering-for-single-cell-data-integration-and-evaluation">CIDER: Meta-Clustering for Single-Cell Data Integration and Evaluation<a class="anchor" aria-label="anchor" href="#cider-meta-clustering-for-single-cell-data-integration-and-evaluation"></a>
</h1></div>
<!-- badges: start --><!-- badges: end -->
<p>Clustering Single-cell RNA-Seq (scRNA-Seq) data from multiple samples or conditions are often challenged by confounding factors, such as batch effects and biologically relevant variability. Existing batch effect removal methods typically require strong assumptions on the composition of cell populations being near identical across samples. Here we present <strong>CIDER</strong>, a <strong>meta-clustering workflow</strong> based on inter-group similarity measures. The prototype of this method is firstly applied in <a href="https://www.sciencedirect.com/science/article/pii/S1535610820300428" class="external-link">Hu et al., Cancer Cell 2020</a>.</p>
<p><strong>For more informtion please see our <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02561-2" class="external-link">publication</a> on Genome Biology (2021).</strong></p>
>>>>>>> Stashed changes
<p>CIDER can:</p>
<ol style="list-style-type: decimal">
<li>address the <strong>clustering</strong> task for confounded scRNA-Seq data, or</li>
<li>assess the biological correctness of integration as <strong>a test metric</strong>, while it does <strong>not</strong> require the existence of prior cellular annotations.</li>
</ol>
<<<<<<< Updated upstream
<p>For more informtion please see our <a href="https://doi.org/10.1101/2021.03.29.437525">preprint</a>.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install CIDER from <a href="https://github.com/zhiyhu/CIDER/">github</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'zhiyhu/CIDER'</span><span class="op">)</span></code></pre></div>
=======
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install CIDER from <a href="https://CRAN.R-project.org/package=CIDER" class="external-link">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"CIDER"</span><span class="op">)</span></code></pre></div>
>>>>>>> Stashed changes
</div>
<div class="section level2">
<h2 id="cider-as-an-evaluation-metric---quick-start">CIDER as an evaluation metric - Quick start<a class="anchor" aria-label="anchor" href="#cider-as-an-evaluation-metric---quick-start"></a>
</h2>
<p>If you have scRNA-Seq data corrected by an integration algorithm (e.g. Seurat-CCA, Harmony, Scanrama…). You can use CIDER to evaluate if the biological populations are correctly aligned.</p>
<p>Before running CIDER evaluation functions, make sure that you have a Seurat object (e.g. <code>seu.integrated</code>) with corrected PCs in <code>seu.integrated@reductions$pca@cell.embeddings</code>. Seurat-CCA automatically put the corrected PCs there. If other methods are used, the corrected PCs can be added using <code>seu.integrated@reductions$pca@cell.embeddings &lt;- corrected.PCs</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyhu/CIDER" class="external-link">CIDER</a></span><span class="op">)</span>
<span class="va">seu.integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/hdbscan.seurat.html">hdbscan.seurat</a></span><span class="op">(</span><span class="va">seu.integrated</span><span class="op">)</span>
<span class="va">ider</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getIDEr.html">getIDEr</a></span><span class="op">(</span><span class="va">seu.integrated</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu.integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/estimateProb.html">estimateProb</a></span><span class="op">(</span><span class="va">seu.integrated</span>, <span class="va">ider</span><span class="op">)</span></code></pre></div>
<p>The evaluation scores (IDER-based similarity and empirical p values) can be visualised by the <code>scatterPlot</code> function. A <a href="https://zhiyhu.github.io/CIDER/articles/evaluation.html" class="external-link">detailed tutorial of evaluation</a> is available.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu.integrated</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"similarity"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu.integrated</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"pvalue"</span><span class="op">)</span> 
<span class="fu">plot_grid</span><span class="op">(</span><span class="va">p1</span>,<span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/evaluation_scatterplot.png"></p>
</div>
<div class="section level2">
<h2 id="use-cider-for-clustering-tasks">Use CIDER for clustering tasks<a class="anchor" aria-label="anchor" href="#use-cider-for-clustering-tasks"></a>
</h2>
<!--- ![](man/figures/clustering_diagram.png)---><div class="section level3">
<h3 id="quick-start---ascider">Quick start - asCIDER<a class="anchor" aria-label="anchor" href="#quick-start---ascider"></a>
</h3>
<p>Here <code>seu</code> is a Seurat object with initial clustering annotation stored in <code>initial_cluster</code> of metadata and batch information in <code>Batch</code>. The asCIDER example here contains two steps: computing IDER-based similarity matrix (<code>getIDER</code>) and performing the final clustering (<code>finalClustering</code>).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ider</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getIDEr.html">getIDEr</a></span><span class="op">(</span><span class="va">seu</span>, 
                group.by.var <span class="op">=</span> <span class="st">"initial_cluster"</span>,
                batch.by.var <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/finalClustering.html">finalClustering</a></span><span class="op">(</span><span class="va">seu</span>, <span class="va">ider</span>, cutree.h <span class="op">=</span> <span class="fl">0.45</span><span class="op">)</span></code></pre></div>
<<<<<<< Updated upstream
<p>A detailed tutorial of asCIDER is <a href="https://zhiyhu.github.io/CIDER/articles/asCIDER.html">here</a>. If your data do not have prior batch-specific clusters, the dnCDIER tutorials (<a href="https://zhiyhu.github.io/CIDER/articles/dnCIDER_highlevel.html">high-level</a> and <a href="https://zhiyhu.github.io/CIDER/articles/dnCIDER.html">detailed walk-through</a>) can be referred to.</p>
<!--- You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this. You could also use GitHub Actions to re-render `README.Rmd` every time you push. An example workflow can be found here: <https://github.com/r-lib/actions/tree/master/examples>.--->
</div>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Z. Hu, A. A. Ahmed, C. Yau. An interpretable meta-clustering framework for single-cell RNA-Seq data integration and evaluation. <em>bioRxiv</em> 2021.03.29.437525; <a href="https://doi.org/10.1101/2021.03.29.437525">doi: https://doi.org/10.1101/2021.03.29.437525</a></p>
<p>Z. Hu, M. Artibani, A. Alsaadi, N. Wietek, M. Morotti, T. Shi, Z. Zhong, L. Santana Gonzalez, S. El-Sahhar, M. KaramiNejadRanjbar, G. Mallett, Y. Feng, K. Masuda, Y. Zheng, K. Chong, S. Damato, S. Dhar, L. Campo, R. Garruto Campanile, V. Rai, D. Maldonado-Perez, S. Jones, V. Cerundolo, T. Sauka-Spengler, C. Yau<em>, A. A. Ahmed</em>. The repertoire of serous ovarian cancer non-genetic heterogeneity revealed by single-cell sequencing of normal fallopian tube epithelial cells. Cancer Cell 37 (2), p226-242.E7 (2020). <a href="https://www.sciencedirect.com/science/article/pii/S1535610820300428">doi: https://doi.org/10.1101/2021.03.29.437525</a></p>
=======
<p>A detailed tutorial of asCIDER is <a href="https://zhiyhu.github.io/CIDER/articles/asCIDER.html" class="external-link">here</a>. If your data do not have prior batch-specific clusters, the dnCDIER tutorials (<a href="https://zhiyhu.github.io/CIDER/articles/dnCIDER_highlevel.html" class="external-link">high-level</a> and <a href="https://zhiyhu.github.io/CIDER/articles/dnCIDER.html" class="external-link">detailed walk-through</a>) can be referred to.</p>
</div>
</div>
<div class="section level2">
<h2 id="quick-start---compute-the-similarity-matrix-within-one-batch">Quick start - compute the similarity matrix within one batch<a class="anchor" aria-label="anchor" href="#quick-start---compute-the-similarity-matrix-within-one-batch"></a>
</h2>
<p>Here is the code used to compute the similarity matrix within one batch.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyhu/CIDER" class="external-link">CIDER</a></span><span class="op">)</span>
<span class="co"># make sure have a colume called "Batch" in the data and assign a uniform value</span>
<span class="co"># to all cells, for example:</span>
<span class="va">seu</span><span class="op">$</span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="st">"onebatch"</span> 
<span class="co"># run the getDistMat, the input needs to be list(seu), the tmp.initial.clusters </span>
<span class="co"># is the group information that you want to compute the similarity matrix with.</span>
<span class="co"># The output is the similarity matrix.</span>
<span class="va">dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getDistMat.html">getDistMat</a></span><span class="op">(</span>seu_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span>, tmp.initial.clusters <span class="op">=</span> <span class="st">"cell_type"</span><span class="op">)</span></code></pre></div>
<p>A more dedicated function is under construction.</p>
<!--- You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this. You could also use GitHub Actions to re-render `README.Rmd` every time you push. An example workflow can be found here: <https://github.com/r-lib/actions/tree/master/examples>.--->
</div>
<div class="section level2">
<h2 id="bug-reports-and-issues">Bug reports and issues<a class="anchor" aria-label="anchor" href="#bug-reports-and-issues"></a>
</h2>
<p>Please use <a href="https://github.com/zhiyhu/CIDER/issues" class="external-link">Issues</a> to report bugs or seek help. Thank you!</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Z. Hu, A. A. Ahmed, C. Yau. CIDER: an interpretable meta-clustering framework for single-cell RNA-seq data integration and evaluation. <em>Genome Biology</em> 22, Article number: 337 (2021); <a href="https://doi.org/10.1186/s13059-021-02561-2" class="external-link">doi: https://doi.org/10.1186/s13059-021-02561-2</a></p>
>>>>>>> Stashed changes
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<<<<<<< Updated upstream
<li>Browse source code at <br><a href="https://github.com/zhiyhu/CIDER/">https://​github.com/​zhiyhu/​CIDER/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/zhiyhu/CIDER/issues">https://​github.com/​zhiyhu/​CIDER/​issues</a>
</li>
=======
<li><a href="https://cloud.r-project.org/package=CIDER" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/zhiyhu/CIDER/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/zhiyhu/CIDER/issues" class="external-link">Report a bug</a></li>
>>>>>>> Stashed changes
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CIDER</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Zhiyuan Hu <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1688-6032" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Christopher Yau <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-7615-8523" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://zenodo.org/badge/latestdoi/296897483" class="external-link"><img src="https://zenodo.org/badge/296897483.svg" alt="DOI"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Zhiyuan Hu, Christopher Yau.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
