<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Start with Assisted CIDER (asCIDER) • CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Start with Assisted CIDER (asCIDER)">
<meta property="og:description" content="Hu et al. An interpretable meta-clustering framework for single-cell RNA-Seq data integration and evaluation">
<meta property="og:image" content="http://zhiyhu.github.io/CIDER/reference/figures/card.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CIDER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a>
    </li>
    <li>
      <a href="../articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a>
    </li>
    <li>
      <a href="../articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER)</a>
    </li>
    <li>
      <a href="../articles/evaluation.html">Evaluating integration results by CIDER</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhiyhu/CIDER/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Start with Assisted CIDER (asCIDER)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhiyhu/CIDER/blob/HEAD/vignettes/asCIDER.Rmd" class="external-link"><code>vignettes/asCIDER.Rmd</code></a></small>
      <div class="hidden name"><code>asCIDER.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette performs asCIDER, a meta-clustering method, on a cross-species pancreas dataset. AsCIDER is aimed to achieve the clustering task in data confounded by unwanted variables. In this scenario, the unwanted variable is species effects.</p>
<p>asCIDER is short for assisted CIDER, assisted by the prior batch-specific annotations or clustering results.</p>
</div>
<div class="section level2">
<h2 id="set-up">Set up<a class="anchor" aria-label="anchor" href="#set-up"></a>
</h2>
<p>In addition to <strong>CIDER</strong>, we will load the following packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyhu/CIDER" class="external-link">CIDER</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pancreas-data">Pancreas data<a class="anchor" aria-label="anchor" href="#pancreas-data"></a>
</h2>
<p>The example data can be downloaded from <a href="https://figshare.com/s/d5474749ca8c711cc205" class="external-link uri">https://figshare.com/s/d5474749ca8c711cc205</a>.</p>
<p>This data<span class="math inline">\(^1\)</span> contain cells from human (8241 cells) and mouse (1886 cells).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_counts.RData"</span><span class="op">)</span> <span class="co"># count matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_meta.RData"</span><span class="op">)</span> <span class="co"># meta data/cell information</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pancreas_counts</span>, meta.data <span class="op">=</span> <span class="va">pancreas_meta</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; human mouse </span>
<span class="co">#&gt;  8241  1886</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exam-if-data-are-confounded">Exam if data are confounded<a class="anchor" aria-label="anchor" href="#exam-if-data-are-confounded"></a>
</h2>
<p>Prior to use CIDER (or other integration methods for clustering) it is important to exam if clustering will be confounded by the cross-species factors.</p>
<p>Her we first perform the conventional Seurat<span class="math inline">\(^2\)</span> clustering pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seu</span>, npcs <span class="op">=</span> <span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="confounded-dimension-reduction">Confounded dimension reduction<a class="anchor" aria-label="anchor" href="#confounded-dimension-reduction"></a>
</h3>
<p>The PCA and <em>t</em>-SNE plots both showed that the data are confounded by species.</p>
<p>The <code>scatterPlot</code> function is used to generate dimension reduction figures. It takes input of a Seurat object (<code>seu</code> here), the name of reduction (<code>pca</code> and <code>tsne</code> here), the variable deciding dot colours (<code>Batch</code> here) and the title of plots. See more information by <code><a href="../reference/scatterPlot.html">?scatterPlot</a></code></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"pca"</span>,colour.by <span class="op">=</span> <span class="st">"Batch"</span>, title <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span> 
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>,colour.by <span class="op">=</span> <span class="st">"Batch"</span>, title <span class="op">=</span> <span class="st">"t-SNE"</span><span class="op">)</span> 
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span></code></pre></div>
<p><img src="asCIDER_files/figure-html/scatterPlot-confounded-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="confounded-clustering-results">Confounded clustering results<a class="anchor" aria-label="anchor" href="#confounded-clustering-results"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">seu</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span>
<span class="co">#&gt; Computing nearest neighbor graph</span>
<span class="co">#&gt; Computing SNN</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span>
<span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Number of nodes: 10127</span>
<span class="co">#&gt; Number of edges: 352494</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Louvain algorithm...</span>
<span class="co">#&gt; Maximum modularity in 10 random starts: 0.9149</span>
<span class="co">#&gt; Number of communities: 22</span>
<span class="co">#&gt; Elapsed time: 0 seconds</span>

<span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>,colour.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, title <span class="op">=</span> <span class="st">"t-SNE"</span><span class="op">)</span> </code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-3-1.png" width="480"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">seurat_clusters</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">Var2</span>, y<span class="op">=</span><span class="va">Freq</span>, x<span class="op">=</span><span class="va">Var1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"CIDER_cluster"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Proportions"</span><span class="op">)</span></code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-4-1.png" width="480"></p>
</div>
</div>
<div class="section level2">
<h2 id="ascider">asCIDER<a class="anchor" aria-label="anchor" href="#ascider"></a>
</h2>
<div class="section level3">
<h3 id="prepare-initial-clusters">Prepare initial clusters<a class="anchor" aria-label="anchor" href="#prepare-initial-clusters"></a>
</h3>
<p>asCIDER uses existing within-batch clustering results. Here we concatenate the batch ID and the within-batch cluster ID to obtain cluster-specific groups (i.e. initial clusters).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span><span class="op">$</span><span class="va">initial_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Group</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">initial_cluster</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             acinar_human activated_stellate_human activated_stellate_mouse </span>
<span class="co">#&gt;                      932                      275                       14 </span>
<span class="co">#&gt;              alpha_human              alpha_mouse             b_cell_mouse </span>
<span class="co">#&gt;                     2241                      191                       10 </span>
<span class="co">#&gt;               beta_human               beta_mouse              delta_human </span>
<span class="co">#&gt;                     2455                      894                      592 </span>
<span class="co">#&gt;              delta_mouse             ductal_human             ductal_mouse </span>
<span class="co">#&gt;                      218                     1033                      275 </span>
<span class="co">#&gt;        endothelial_human        endothelial_mouse            epsilon_human </span>
<span class="co">#&gt;                      214                      139                       17 </span>
<span class="co">#&gt;              gamma_human              gamma_mouse       immune_other_mouse </span>
<span class="co">#&gt;                      241                       41                        8 </span>
<span class="co">#&gt;         macrophage_human         macrophage_mouse               mast_human </span>
<span class="co">#&gt;                       39                       36                       25 </span>
<span class="co">#&gt; quiescent_stellate_human quiescent_stellate_mouse            schwann_human </span>
<span class="co">#&gt;                      160                       47                       10 </span>
<span class="co">#&gt;            schwann_mouse             t_cell_human             t_cell_mouse </span>
<span class="co">#&gt;                        6                        7                        7</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="calculate-of-ider-similarity-matrix">Calculate of IDER similarity matrix<a class="anchor" aria-label="anchor" href="#calculate-of-ider-similarity-matrix"></a>
</h3>
<p>The function <code>getIDEr</code> calculate the IDER-based distance matrix.</p>
<p>By default, it will use the column called “initial_cluster” as initial clusters, and “Batch” as batch. If you are using columns other than this two, please revise these two parameters.</p>
<p>For this step, you can choose to use parallel computation by setting <code>use.parallel = TRUE</code> or not (default <code>use.parallel = FALSE</code>). The default number of cores used for parallel computation is <code>detectCores(logical = FALSE) - 1</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ider</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getIDEr.html">getIDEr</a></span><span class="op">(</span><span class="va">seu</span>, 
                group.by.var <span class="op">=</span> <span class="st">"initial_cluster"</span>,
                batch.by.var <span class="op">=</span> <span class="st">"Batch"</span>,
                downsampling.size <span class="op">=</span> <span class="fl">35</span>, 
                use.parallel <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualise-the-similarity-matrix">Visualise the similarity matrix<a class="anchor" aria-label="anchor" href="#visualise-the-similarity-matrix"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>, <span class="st">"gamma"</span>,<span class="st">"ductal"</span>,<span class="st">"endothelial"</span>, <span class="st">"activated_stellate"</span>, <span class="st">"quiescent_stellate"</span>, <span class="st">"macrophage"</span><span class="op">)</span>
<span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">groups</span>, <span class="st">"_human"</span><span class="op">)</span>
<span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">groups</span>, <span class="st">"_mouse"</span><span class="op">)</span>

<span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span>
  <span class="va">ider</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">]</span>,
  color <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">inferno</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,
  border_color <span class="op">=</span> <span class="cn">NA</span>,
  display_numbers <span class="op">=</span> <span class="cn">TRUE</span>,
  cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
  width <span class="op">=</span> <span class="fl">7</span>,
  height <span class="op">=</span> <span class="fl">5</span>,
  cellwidth <span class="op">=</span> <span class="fl">22</span>,
  cellheight <span class="op">=</span> <span class="fl">22</span>
<span class="op">)</span></code></pre></div>
<p><img src="asCIDER_files/figure-html/heatmap-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="perform-final-clustering">Perform final Clustering<a class="anchor" aria-label="anchor" href="#perform-final-clustering"></a>
</h3>
<p>Next we put both Seurat object and the similarity matrix list into <code>finalClustering</code>. You can either cut trees by height (set <code>cutree.by = 'h</code>) or by k (set <code>cutree.by = 'k</code>). The default is cutting by the height of 0.45.</p>
<p>The final clustering results are stored in Colume <code>cider_final_cluster</code> of Seurat object metadata and can be extracted using <code>seu$cider_final_cluster</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finalClustering.html">finalClustering</a></span><span class="op">(</span><span class="va">seu</span>, <span class="va">ider</span>, cutree.h <span class="op">=</span> <span class="fl">0.45</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>
<span class="co">#&gt;                             orig.ident nCount_RNA nFeature_RNA Batch   Group</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0001     mouse1      11003         2994 mouse    beta</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0002     mouse1      10701         3565 mouse  ductal</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0003     mouse1       9445         2401 mouse   delta</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0004     mouse1       8193         2865 mouse schwann</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0005     mouse1       7323         2355 mouse   delta</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0006     mouse1       8504         2637 mouse    beta</span>
<span class="co">#&gt;                             Sample RNA_snn_res.0.8 seurat_clusters</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0001 mouse1               9               9</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0002 mouse1              13              13</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0003 mouse1               7               7</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0004 mouse1              21              21</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0005 mouse1               7               7</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0006 mouse1               8               8</span>
<span class="co">#&gt;                             initial_cluster CIDER_cluster</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0001      beta_mouse             1</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0002    ductal_mouse             2</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0003     delta_mouse             3</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0004   schwann_mouse             4</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0005     delta_mouse             3</span>
<span class="co">#&gt; mouse1_lib1.final_cell_0006      beta_mouse             1</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualise-clustering-results">Visualise clustering results<a class="anchor" aria-label="anchor" href="#visualise-clustering-results"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">plot_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"CIDER_cluster"</span>, title <span class="op">=</span> <span class="st">"asCIDER clustering results"</span><span class="op">)</span> 
<span class="va">plot_list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"Group"</span>, title <span class="op">=</span> <span class="st">"Ground truth of cell populations"</span><span class="op">)</span> 
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plot_list</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="asCIDER_files/figure-html/summary-tsne-plots-1.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">CIDER_cluster</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">Var2</span>, y<span class="op">=</span><span class="va">Freq</span>, x<span class="op">=</span><span class="va">Var1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"CIDER_cluster"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Proportions"</span><span class="op">)</span></code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-5-1.png" width="480"></p>
</div>
<div class="section level2">
<h2 id="technical">Technical<a class="anchor" aria-label="anchor" href="#technical"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] viridis_0.6.2      viridisLite_0.4.0  cowplot_1.1.1      ggplot2_3.3.5     </span>
<span class="co">#&gt; [5] pheatmap_1.0.12    SeuratObject_4.0.4 Seurat_4.1.0       CIDER_0.99.0      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] systemfonts_1.0.2     plyr_1.8.6            igraph_1.2.8         </span>
<span class="co">#&gt;   [4] lazyeval_0.2.2        splines_4.1.2         listenv_0.8.0        </span>
<span class="co">#&gt;   [7] scattermore_0.7       digest_0.6.28         foreach_1.5.1        </span>
<span class="co">#&gt;  [10] htmltools_0.5.2       fansi_0.5.0           magrittr_2.0.1       </span>
<span class="co">#&gt;  [13] memoise_2.0.0         tensor_1.5            cluster_2.1.2        </span>
<span class="co">#&gt;  [16] doParallel_1.0.16     ROCR_1.0-11           limma_3.50.0         </span>
<span class="co">#&gt;  [19] globals_0.14.0        matrixStats_0.61.0    pkgdown_2.0.6        </span>
<span class="co">#&gt;  [22] spatstat.sparse_2.0-0 colorspace_2.0-2      ggrepel_0.9.1        </span>
<span class="co">#&gt;  [25] textshaping_0.3.6     xfun_0.28             dplyr_1.0.7          </span>
<span class="co">#&gt;  [28] crayon_1.4.2          jsonlite_1.7.2        spatstat.data_2.1-0  </span>
<span class="co">#&gt;  [31] survival_3.2-13       zoo_1.8-9             iterators_1.0.13     </span>
<span class="co">#&gt;  [34] glue_1.5.0            polyclip_1.10-0       gtable_0.3.0         </span>
<span class="co">#&gt;  [37] leiden_0.3.9          kernlab_0.9-29        future.apply_1.8.1   </span>
<span class="co">#&gt;  [40] abind_1.4-5           scales_1.1.1          DBI_1.1.1            </span>
<span class="co">#&gt;  [43] edgeR_3.36.0          miniUI_0.1.1.1        Rcpp_1.0.7           </span>
<span class="co">#&gt;  [46] xtable_1.8-4          reticulate_1.22       spatstat.core_2.3-1  </span>
<span class="co">#&gt;  [49] htmlwidgets_1.5.4     httr_1.4.2            RColorBrewer_1.1-2   </span>
<span class="co">#&gt;  [52] ellipsis_0.3.2        ica_1.0-2             farver_2.1.0         </span>
<span class="co">#&gt;  [55] pkgconfig_2.0.3       sass_0.4.0            uwot_0.1.10          </span>
<span class="co">#&gt;  [58] deldir_1.0-6          locfit_1.5-9.4        utf8_1.2.2           </span>
<span class="co">#&gt;  [61] labeling_0.4.2        tidyselect_1.1.1      rlang_1.0.5          </span>
<span class="co">#&gt;  [64] reshape2_1.4.4        later_1.3.0           munsell_0.5.0        </span>
<span class="co">#&gt;  [67] tools_4.1.2           cachem_1.0.6          cli_3.1.0            </span>
<span class="co">#&gt;  [70] dbscan_1.1-8          generics_0.1.1        ggridges_0.5.3       </span>
<span class="co">#&gt;  [73] evaluate_0.14         stringr_1.4.0         fastmap_1.1.0        </span>
<span class="co">#&gt;  [76] yaml_2.2.1            ragg_1.1.3            goftest_1.2-3        </span>
<span class="co">#&gt;  [79] knitr_1.36            fs_1.5.0              fitdistrplus_1.1-6   </span>
<span class="co">#&gt;  [82] purrr_0.3.4           RANN_2.6.1            pbapply_1.5-0        </span>
<span class="co">#&gt;  [85] future_1.23.0         nlme_3.1-153          mime_0.12            </span>
<span class="co">#&gt;  [88] compiler_4.1.2        rstudioapi_0.13       plotly_4.10.0        </span>
<span class="co">#&gt;  [91] png_0.1-7             spatstat.utils_2.2-0  tibble_3.1.6         </span>
<span class="co">#&gt;  [94] bslib_0.3.1           stringi_1.7.5         highr_0.9            </span>
<span class="co">#&gt;  [97] desc_1.4.0            lattice_0.20-45       Matrix_1.3-4         </span>
<span class="co">#&gt; [100] vctrs_0.3.8           pillar_1.6.4          lifecycle_1.0.1      </span>
<span class="co">#&gt; [103] spatstat.geom_2.3-0   lmtest_0.9-39         jquerylib_0.1.4      </span>
<span class="co">#&gt; [106] RcppAnnoy_0.0.19      data.table_1.14.2     irlba_2.3.3          </span>
<span class="co">#&gt; [109] httpuv_1.6.3          patchwork_1.1.1       R6_2.5.1             </span>
<span class="co">#&gt; [112] promises_1.2.0.1      KernSmooth_2.23-20    gridExtra_2.3        </span>
<span class="co">#&gt; [115] parallelly_1.28.1     codetools_0.2-18      MASS_7.3-54          </span>
<span class="co">#&gt; [118] assertthat_0.2.1      rprojroot_2.0.2       withr_2.5.0          </span>
<span class="co">#&gt; [121] sctransform_0.3.3     mgcv_1.8-38           parallel_4.1.2       </span>
<span class="co">#&gt; [124] grid_4.1.2            rpart_4.1-15          tidyr_1.1.4          </span>
<span class="co">#&gt; [127] rmarkdown_2.11        Rtsne_0.15            shiny_1.7.1</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Baron, M. et al. A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter- and Intra-cell Population Structure. Cell Syst 3, 346–360.e4 (2016).</li>
<li>Satija R, et al. Spatial reconstruction of single-cell gene expression data. Nature Biotechnology 33, 495-502 (2015).</li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zhiyuan Hu, Christopher Yau.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
