<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Start with De Novo CIDER (dnCIDER) • CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Start with De Novo CIDER (dnCIDER)">
<meta property="og:description" content="CIDER">
<meta property="og:image" content="http://zhiyhu.github.io/CIDER/reference/figures/card.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CIDER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a>
    </li>
    <li>
      <a href="../articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a>
    </li>
    <li>
      <a href="../articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER)</a>
    </li>
    <li>
      <a href="../articles/evaluation.html">Evaluating integration results by CIDER</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhiyhu/CIDER/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Start with De Novo CIDER (dnCIDER)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhiyhu/CIDER/blob/HEAD/vignettes/dnCIDER_highlevel.Rmd" class="external-link"><code>vignettes/dnCIDER_highlevel.Rmd</code></a></small>
      <div class="hidden name"><code>dnCIDER_highlevel.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette performs dnCIDER on a cross-species pancreas dataset.</p>
</div>
<div class="section level2">
<h2 id="set-up">Set up<a class="anchor" aria-label="anchor" href="#set-up"></a>
</h2>
<p>In addition to <strong>CIDER</strong>, we will load the following packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyhu/CIDER" class="external-link">CIDER</a></span><span class="op">)</span>
<span class="co">#&gt; Registered S3 method overwritten by 'spatstat.geom':</span>
<span class="co">#&gt;   method     from</span>
<span class="co">#&gt;   print.boxx cli</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-pancreas-data">Load pancreas data<a class="anchor" aria-label="anchor" href="#load-pancreas-data"></a>
</h2>
<p>The example data can be downloaded from <a href="https://figshare.com/s/d5474749ca8c711cc205" class="external-link uri">https://figshare.com/s/d5474749ca8c711cc205</a>.</p>
<p>Pancreatic cell data<span class="math inline">\(^1\)</span> contain cells from human (8241 cells) and mouse (1886 cells).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_counts.RData"</span><span class="op">)</span> <span class="co"># count matrix</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_meta.RData"</span><span class="op">)</span> <span class="co"># meta data/cell information</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pancreas_counts</span>, meta.data <span class="op">=</span> <span class="va">pancreas_meta</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; human mouse </span>
<span class="co">#&gt;  8241  1886</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="perform-dncider-high-level">Perform dnCIDER (high-level)<a class="anchor" aria-label="anchor" href="#perform-dncider-high-level"></a>
</h2>
<p>DnCIDER contains three steps</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>seu <span class="ot">&lt;-</span> <span class="fu">initialClustering</span>(seu, <span class="at">additional.vars.to.regress =</span> <span class="st">"Sample"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================</span>                                   <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>ider <span class="ot">&lt;-</span> <span class="fu">getIDEr</span>(seu, <span class="at">downsampling.size =</span> <span class="dv">35</span>, <span class="at">use.parallel =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>seu <span class="ot">&lt;-</span> <span class="fu">finalClustering</span>(seu, ider, <span class="at">cutree.h =</span> <span class="fl">0.35</span>) <span class="co"># final clustering</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualise-clustering-results">Visualise clustering results<a class="anchor" aria-label="anchor" href="#visualise-clustering-results"></a>
</h2>
<p>We use the Seurat pipeline to perform normalisation (<code>NormalizeData</code>), preprocessing (<code>FindVariableFeatures</code> and <code>ScaleData</code>) and dimension reduction (<code>RunPCA</code> and <code>RunTSNE</code>).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seu</span>, npcs <span class="op">=</span> <span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span></code></pre></div>
<p>We can see</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"CIDER_cluster"</span>, title <span class="op">=</span> <span class="st">"asCIDER clustering results"</span><span class="op">)</span> </code></pre></div>
<p><img src="dnCIDER_highlevel_files/figure-html/tsne-plot-CIDER-results-1.png" width="384"></p>
<p>By comparing the dnCIDER results to the cell annotation from the publication<span class="math inline">\(^1\)</span>, we observe that dnCIDER correctly identify the majority of populations across two species.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"Group"</span>, title <span class="op">=</span> <span class="st">"Ground truth of cell populations"</span><span class="op">)</span> </code></pre></div>
<p><img src="dnCIDER_highlevel_files/figure-html/tsne-plot-ground-truth-1.png" width="384"></p>
</div>
<div class="section level2">
<h2 id="technical">Technical<a class="anchor" aria-label="anchor" href="#technical"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] cowplot_1.1.1      SeuratObject_4.0.4 Seurat_4.1.0       CIDER_0.99.0      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] systemfonts_1.0.2     plyr_1.8.6            igraph_1.2.8         </span>
<span class="co">#&gt;   [4] lazyeval_0.2.2        splines_4.1.2         listenv_0.8.0        </span>
<span class="co">#&gt;   [7] scattermore_0.7       ggplot2_3.3.5         digest_0.6.28        </span>
<span class="co">#&gt;  [10] foreach_1.5.1         htmltools_0.5.2       viridis_0.6.2        </span>
<span class="co">#&gt;  [13] fansi_0.5.0           magrittr_2.0.1        memoise_2.0.0        </span>
<span class="co">#&gt;  [16] tensor_1.5            cluster_2.1.2         doParallel_1.0.16    </span>
<span class="co">#&gt;  [19] ROCR_1.0-11           limma_3.50.0          globals_0.14.0       </span>
<span class="co">#&gt;  [22] matrixStats_0.61.0    pkgdown_2.0.6         spatstat.sparse_2.0-0</span>
<span class="co">#&gt;  [25] colorspace_2.0-2      ggrepel_0.9.1         textshaping_0.3.6    </span>
<span class="co">#&gt;  [28] xfun_0.28             dplyr_1.0.7           crayon_1.4.2         </span>
<span class="co">#&gt;  [31] jsonlite_1.7.2        spatstat.data_2.1-0   survival_3.2-13      </span>
<span class="co">#&gt;  [34] zoo_1.8-9             iterators_1.0.13      glue_1.5.0           </span>
<span class="co">#&gt;  [37] polyclip_1.10-0       gtable_0.3.0          leiden_0.3.9         </span>
<span class="co">#&gt;  [40] kernlab_0.9-29        future.apply_1.8.1    abind_1.4-5          </span>
<span class="co">#&gt;  [43] scales_1.1.1          pheatmap_1.0.12       DBI_1.1.1            </span>
<span class="co">#&gt;  [46] edgeR_3.36.0          miniUI_0.1.1.1        Rcpp_1.0.7           </span>
<span class="co">#&gt;  [49] viridisLite_0.4.0     xtable_1.8-4          reticulate_1.22      </span>
<span class="co">#&gt;  [52] spatstat.core_2.3-1   htmlwidgets_1.5.4     httr_1.4.2           </span>
<span class="co">#&gt;  [55] RColorBrewer_1.1-2    ellipsis_0.3.2        ica_1.0-2            </span>
<span class="co">#&gt;  [58] farver_2.1.0          pkgconfig_2.0.3       sass_0.4.0           </span>
<span class="co">#&gt;  [61] uwot_0.1.10           deldir_1.0-6          locfit_1.5-9.4       </span>
<span class="co">#&gt;  [64] utf8_1.2.2            labeling_0.4.2        tidyselect_1.1.1     </span>
<span class="co">#&gt;  [67] rlang_1.0.5           reshape2_1.4.4        later_1.3.0          </span>
<span class="co">#&gt;  [70] munsell_0.5.0         tools_4.1.2           cachem_1.0.6         </span>
<span class="co">#&gt;  [73] cli_3.1.0             dbscan_1.1-8          generics_0.1.1       </span>
<span class="co">#&gt;  [76] ggridges_0.5.3        evaluate_0.14         stringr_1.4.0        </span>
<span class="co">#&gt;  [79] fastmap_1.1.0         yaml_2.2.1            ragg_1.1.3           </span>
<span class="co">#&gt;  [82] goftest_1.2-3         knitr_1.36            fs_1.5.0             </span>
<span class="co">#&gt;  [85] fitdistrplus_1.1-6    purrr_0.3.4           RANN_2.6.1           </span>
<span class="co">#&gt;  [88] pbapply_1.5-0         future_1.23.0         nlme_3.1-153         </span>
<span class="co">#&gt;  [91] mime_0.12             compiler_4.1.2        rstudioapi_0.13      </span>
<span class="co">#&gt;  [94] plotly_4.10.0         png_0.1-7             spatstat.utils_2.2-0 </span>
<span class="co">#&gt;  [97] tibble_3.1.6          bslib_0.3.1           stringi_1.7.5        </span>
<span class="co">#&gt; [100] highr_0.9             desc_1.4.0            lattice_0.20-45      </span>
<span class="co">#&gt; [103] Matrix_1.3-4          vctrs_0.3.8           pillar_1.6.4         </span>
<span class="co">#&gt; [106] lifecycle_1.0.1       spatstat.geom_2.3-0   lmtest_0.9-39        </span>
<span class="co">#&gt; [109] jquerylib_0.1.4       RcppAnnoy_0.0.19      data.table_1.14.2    </span>
<span class="co">#&gt; [112] irlba_2.3.3           httpuv_1.6.3          patchwork_1.1.1      </span>
<span class="co">#&gt; [115] R6_2.5.1              promises_1.2.0.1      KernSmooth_2.23-20   </span>
<span class="co">#&gt; [118] gridExtra_2.3         parallelly_1.28.1     codetools_0.2-18     </span>
<span class="co">#&gt; [121] MASS_7.3-54           assertthat_0.2.1      rprojroot_2.0.2      </span>
<span class="co">#&gt; [124] sctransform_0.3.3     mgcv_1.8-38           grid_4.1.2           </span>
<span class="co">#&gt; [127] rpart_4.1-15          tidyr_1.1.4           rmarkdown_2.11       </span>
<span class="co">#&gt; [130] Rtsne_0.15            shiny_1.7.1</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Baron, M. et al. A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter- and Intra-cell Population Structure. Cell Syst 3, 346–360.e4 (2016).</li>
<li>Satija R, et al. Spatial reconstruction of single-cell gene expression data. Nature Biotechnology 33, 495-502 (2015).</li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zhiyuan Hu, Christopher Yau.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
