<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Evaluating integration results by CIDER • CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Evaluating integration results by CIDER">
<meta property="og:description" content="CIDER">
<meta property="og:image" content="http://zhiyhu.github.io/CIDER/reference/figures/card.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CIDER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a>
    </li>
    <li>
      <a href="../articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a>
    </li>
    <li>
      <a href="../articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER)</a>
    </li>
    <li>
      <a href="../articles/evaluation.html">Evaluating integration results by CIDER</a>
    </li>
    <li>
      <a href="../articles/evaluation_obsolete.html">Evaluating integration results by CIDER</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhiyhu/CIDER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="evaluation_obsolete_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Evaluating integration results by CIDER</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhiyhu/CIDER/blob/master/vignettes/evaluation_obsolete.Rmd"><code>vignettes/evaluation_obsolete.Rmd</code></a></small>
      <div class="hidden name"><code>evaluation_obsolete.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Integration and batch correction methods have become a <strong>popular</strong> component in the bioinformatic workflows for scRNA-Seq data analysis, whilst the integration results (mostly corrected PCs or less commonly corrected read counts) are <strong>rarely</strong> validated or evaluated with an objective metric.</p>
<p>To assess the correctness of integration (i.e. whether cells belonging to the same population are gathered and ones belonging to different populations stay separate after integration), the existing evaluation metrics require the existence of ground truth for cell population annotations.</p>
<p>CIDER provides a <strong>ground-truth-free</strong> approach to evaluate the integration results. This vignette focuses how showing the process using the example data of dendritic cells.</p>
</div>
<div id="set-up" class="section level1">
<h1 class="hasAnchor">
<a href="#set-up" class="anchor"></a>Set up</h1>
<p>Apart from <strong>CIDER</strong>, the following packages also need to be loaded:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyhu/CIDER">CIDER</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="load-dendritic-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-dendritic-data" class="anchor"></a>Load dendritic data</h1>
<p>The example data are loaded using <code>data("dendritic")</code>. It contains 26593 genes and 564 cells from two batches.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># data("dendritic")</span>
<span class="co"># dim(dendritic)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># table(dendritic$Batch)</span></code></pre></div>
</div>
<div id="perform-integration" class="section level1">
<h1 class="hasAnchor">
<a href="#perform-integration" class="anchor"></a>Perform integration</h1>
<p>First an integration method<span class="math inline">\(^1\)</span> is applied on the dendritic data. You can apply other integration methods to the your data, as long as the correct PCs are stored in your Seurat object, i.e. <code>Reductions(seu.integrated, "pca")</code> or <code>seu.integrated@reductions$pca</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># seu.list &lt;- SplitObject(dendritic, split.by = "Batch")</span>
<span class="co"># for (i in 1:length(seu.list)) {</span>
<span class="co">#   seu.list[[i]] &lt;- NormalizeData(seu.list[[i]], verbose = FALSE)</span>
<span class="co">#   seu.list[[i]] &lt;- FindVariableFeatures(seu.list[[i]], </span>
<span class="co">#                                         selection.method = "vst",</span>
<span class="co">#                                         nfeatures = 1000, verbose = FALSE)</span>
<span class="co"># }</span>
<span class="co"># seu.anchors &lt;- FindIntegrationAnchors(object.list = seu.list, </span>
<span class="co">#                                       dims = 1:15, verbose = FALSE)</span>
<span class="co"># seu.integrated &lt;- IntegrateData(anchorset = seu.anchors, </span>
<span class="co">#                                 dims = 1:15, verbose = FALSE)</span>
<span class="co"># </span>
<span class="co"># DefaultAssay(seu.integrated) &lt;- "integrated"</span>
<span class="co"># seu.integrated &lt;- ScaleData(seu.integrated, verbose = FALSE)</span>
<span class="co"># seu.integrated &lt;- RunPCA(seu.integrated, verbose = FALSE)</span>
<span class="co"># seu.integrated &lt;- RunTSNE(seu.integrated, reduction = "pca", dims = 1:5)</span>
<span class="co"># seu.integrated@reductions$pca</span></code></pre></div>
<p>Clear the intermediate outcome.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># rm(seu.list, seu.anchors)</span>
<span class="co"># gc()</span></code></pre></div>
</div>
<div id="evaluate-by-cider" class="section level1">
<h1 class="hasAnchor">
<a href="#evaluate-by-cider" class="anchor"></a>Evaluate by CIDER</h1>
<div id="calculate-similarity-and-p-values-essential" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-similarity-and-p-values-essential" class="anchor"></a>Calculate similarity and p values (essential)</h2>
<p>CIDER evaluates integration results in three steps:</p>
<ol style="list-style-type: decimal">
<li>Clustering based on the corrected PCs (<code>hdbscan.seurat</code>). This step uses HDBSCAN, which is a density-based clustering algorithm<span class="math inline">\(^2\)</span>. The clustering results are stored in <code>seu.integrated$dbscan_cluster</code>. Clusters are further divided into batch-specific clusters by concatenating dbscan_cluster and batch, stored in <code>seu.integrated$initial_cluster</code>.</li>
<li>Compute IDER-based similarity matrix (<code>getIDEr</code>) among the batch-specific initial clusters. If multiple CPUs are availble, you can set <code>use.parallel = TRUE</code> and <code>n.cores</code> to the number of available cores to speed it up.</li>
<li>Assign the similarity and estimate empirical p values (<code>estimateProb</code>) for the correctness of integration. High similarity values and low p values indicate that the cell are similar to the surrounding cells and likely integrated correctly.</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># seu.integrated &lt;- hdbscan.seurat(seu.integrated)</span>
<span class="co"># ider &lt;- getIDEr(seu.integrated, use.parallel = FALSE, verbose = FALSE)</span>
<span class="co"># seu.integrated &lt;- estimateProb(seu.integrated, ider)</span></code></pre></div>
</div>
<div id="evaluation-scores" class="section level2">
<h2 class="hasAnchor">
<a href="#evaluation-scores" class="anchor"></a>Evaluation scores</h2>
<p>The evaluation scores can be viewed by the <code>scatterPlot</code> as below. As shown cells with dbscan_cluster of 2 and 3 have low regional similarity and high empirical p values, suggesting that they can be incorrectly integrated.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># p1 &lt;- scatterPlot(seu.integrated, "tsne", "dbscan_cluster")</span>
<span class="co"># p2 &lt;- scatterPlot(seu.integrated, "tsne", colour.by = "similarity") + </span>
<span class="co">#   labs(fill = "Similarity")</span>
<span class="co"># p3 &lt;- scatterPlot(seu.integrated, "tsne", colour.by = "pvalue") + </span>
<span class="co">#   labs(fill = "Empirical\n p values")</span>
<span class="co"># plot_grid(p1,p2,p3, ncol = 3)</span></code></pre></div>
</div>
<div id="the-ider-based-similarity-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#the-ider-based-similarity-matrix" class="anchor"></a>The IDER-based similarity matrix</h2>
<p>To have more insight, we can view the IDER-based similarity matrix by functions <code>plotNetwork</code> or <code>plotHeatmap</code>. Both of them require the input of a Seurat object and the output of <code>getIDEr</code>. In this example, 1_Batch1 and 1_Batch2 as well as 4_Batch1 and 4_Batch2 have high similarity.</p>
<p><code>plotNetwork</code> generates a graph where vertexes are initial clusters and edge widths are similarity values. The parameter <code>weight.factor</code> controls the scale of edge widths; larger <code>weight.factor</code> will give bolder edges proportionally.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plotNetwork(seu.integrated, ider, weight.factor = 3)</span></code></pre></div>
<p><code>plotHeatmap</code> generates a heatmap where each cell is coloured and labeled by the similarity values.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plotHeatmap(seu.integrated, ider)</span></code></pre></div>
</div>
</div>
<div id="refering-to-ground-truth-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#refering-to-ground-truth-annotation" class="anchor"></a>Refering to ground-truth annotation</h1>
<p>So far the evaluation have completed and CIDER has not used the ground truth at all!</p>
<p>Let’s peep at the ground truth before the closure of this vignette. As shown in the figure below, the clusters having low IDER-based similarity and high p values actually have at least two popuplations (CD1C and CD141), verifying that CIDER spots the wrongly integrated cells.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># scatterPlot(seu.integrated, "tsne", colour.by = "Group") + </span>
<span class="co">#   labs(fill = "Group\n (ground truth)")</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ol style="list-style-type: decimal">
<li>Stuart and Butler et al. Comprehensive Integration of Single-Cell Data. Cell (2019).</li>
<li>Campello, Ricardo JGB, Davoud Moulavi, and Jörg Sander. “Density-based clustering based on hierarchical density estimates.” Pacific-Asia conference on knowledge discovery and data mining. Springer, Berlin, Heidelberg, 2013.</li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Zhiyuan Hu, Christopher Yau.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
